### PR TITLE
Move dev-machine Cloudformation template to the same repo

### PR DESCRIPTION
**Pull request by coffeeavax**
https://github.com/ava-labs/avalanche-ops/issues/371

### CODE REVIEW

File: avalanche-ops/src/dev-machines/artifacts.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    struct Asset;
+    let f = Asset::get("src/aws/cfn-templates/vpc.yaml").unwrap();
+    let s = std::str::from_utf8(f.data.as_ref()).map_err(|e| {
+        Error::new(
+            ErrorKind::InvalidInput,
+            format!("failed to convert embed file to str {}", e),
+        )
+    })?;
+    Ok(s.to_string())
+}
```
Comment (exdx) [inline avalanche-ops/src/dev-machines/artifacts.rs:48]: nit: newline

File: avalanche-ops/src/dev-machines/cfn-templates/asg_ubuntu.yaml [hunk 2]
```diff
@@ -0,0 +1,750 @@
...
+                - !Ref AWS::NoValue
+            - Fn::If:
+                - Has10InstanceTypes
+                - InstanceType: !Select [9, !Ref InstanceTypes]
+                - !Ref AWS::NoValue
+
+Outputs:
+  # same as "AutoScalingGroupName"
+  AsgLogicalId:
+    Value: !Ref ASG
```
Comment (exdx) [inline avalanche-ops/src/dev-machines/cfn-templates/asg_ubuntu.yaml:750]: nit: newline

**Review by exdx — COMMENTED** [review]
This is just a copy/paste from a private repo? Maybe we could add a test or two to ensure everything works as expected? 

I'll leave approval to Gyuho as I'm not familiar with this code at all.

File: avalanche-ops/src/dev-machines/cfn-templates/asg_ubuntu.yaml [hunk 3]
```diff
@@ -0,0 +1,750 @@
...
+
+              export GOPATH=/home/ubuntu/go
+              export PATH=/usr/local/go/bin:/home/ubuntu/go/bin:/home/ubuntu/.cargo/bin:$PATH
+              . /opt/rust/env
+
+              EOF
+
+              # install go
+              sudo rm -rf /usr/local/go
+              GO_VERSION=1.20.4
```
Comment (gyuho) [inline avalanche-ops/src/dev-machines/cfn-templates/asg_ubuntu.yaml:456]:
```suggestion
              GO_VERSION=1.20.5
```

File: avalanche-ops/src/dev-machines/artifacts.rs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+pub fn ec2_instance_role_yaml() -> io::Result<String> {
```
Comment (gyuho) [inline avalanche-ops/src/dev-machines/artifacts.rs:20]: We share the templates with the node asgs, so we don't need these?

File: avalanche-ops/src/dev-machines/artifacts.rs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    let s = std::str::from_utf8(f.data.as_ref()).map_err(|e| {
+        Error::new(
+            ErrorKind::InvalidInput,
+            format!("failed to convert embed file to str {}", e),
+        )
+    })?;
+    Ok(s.to_string())
+}
+
+pub fn vpc_yaml() -> io::Result<String> {
```
Comment (gyuho) [inline avalanche-ops/src/dev-machines/artifacts.rs:35]: we also share vpc with node, so we don't need this

**Pull request merged**